containers_test.go

```diff
@@ -201,3 +202,78 @@ func TestUpload(t *testing.T) {
 		t.Errorf("%s not found", path)
 	}
 }
+
+func TestSquashImage(t *testing.T) {
+	client := DockerClient()
+	if client == nil {
+		t.Skip("Could not find Docker daemon connection")
+	}
+
+	repo := "yourbase-layer-test"
+	tag := "v1"
+	image := fmt.Sprintf("%s:%s", repo, tag)
+
+	buildLayeredImage(t, image)
+
+	err := PullImageIfNotHere(ContainerDefinition{
+		Image: image,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	squashImageId, err := imageId(repo, tag)
+	if err != nil {
+		t.Error(err)
+	}
+
+	layers, _ := CountLayersInImage(squashImageId)
+	if layers <= 1 {
+		t.Error("yourbase-layer-test should have more than one layer.")
+	}
+
+	t.Logf("Pre layer count: %d", layers)
+
+	err = SquashImage(context.Background(), repo, tag)
+	if err != nil {
+		t.Errorf("SquashImage failed: %v", err)
+	}
+
+	newImageId, err := imageId(repo, tag)
+	if err != nil {
+		t.Errorf("Couldn't find squashed image: %s", image)
+	}
+
+	// The new image should have only one layer
+	layers, err = CountLayersInImage(newImageId)
+	if err == nil && layers != 1 {
+		t.Error("yourbase-layer-test should be comprised of a single layer.")
+	}
+
+	t.Logf("Post layer count: %d", layers)
+
+	// Clean up
+	if err := client.RemoveImage(newImageId); err != nil {
+		t.Errorf("Could not removed squashed image")
+	}
+}
+
+func buildLayeredImage(t *testing.T, imageName string) {
+	client := DockerClient()
+
+	dockerFile := []byte("FROM alpine\nRUN apk add curl")
+	size := int64(len(dockerFile))
+	inputbuf, outputbuf := bytes.NewBuffer(nil), bytes.NewBuffer(nil)
```

Comment:
nit: `bytes.NewBuffer` is intended to be used to take ownership of a byte slice. When a new destination buffer is needed, the zero value is the right choice.

```suggestion
	inputbuf, outputbuf := new(bytes.Buffer), new(bytes.Buffer)
```